src/js/services/moonpay/moonpay.service.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+(function(){
+
+angular
+  .module('bitcoincom.services')
+  .factory('moonPayService', moonPayService);
+  
+  function moonPayService(
+    moonPayApiService,
+    localStorageService,
+    $log, $q
+  ) {
+
+    var customerKey = 'moonPayCustomer'
+    var currentCustomer = null;
+    var currentCards = null;
+
+    var service = {
+      // Variables
+
+      // Functions
+      createCustomer: createCustomer,
+      getCustomer: getCustomer,
+      getCustomerId: getCustomerId,
+      updateCustomer: updateCustomer,
+      createCard: createCard,
+      getCards: getCards,
+      getRates: getRates
+    };
+
+    return service;
+
+    /**
+     * Create a customer
+     * @param {String} email 
+     */
+    function createCustomer(email) {
+      // Create the promise
+      var deferred = $q.defer();
+
+      moonPayApiService.createCustomer(email).then(function (customer) {
+        localStorageService.set(customerKey, customer, function (err) {
+          if (err) {
+            $log.debug('Error setting moonpay customer in the local storage');
+            deferred.reject(err);
+          } else {
+            currentCustomer = customer;
+            deferred.resolve(customer);
+          }
+        });
+
+      }, function (err) {
+        $log.debug('Error creating moonpay customer from the api');
+        deferred.reject(err);
+      });
+
+      return deferred.promise;
+    }
+
+    /**
+     * Get customer
+     */
+    function getCustomer() {
+      // Create the promise
+      var deferred = $q.defer();
+
+      // Get the customer in the localStorageService if we didn't do it yet
+      if (currentCustomer != null) {
+        deferred.resolve(currentCustomer);
+      } else {
+        localStorageService.get(customerKey, function (err, customer) {
+          if (err) {
+            $log.debug('Error getting moonpay customer in the local storage');
+            deferred.reject(err);
+          } else {
+            currentCustomer = customer;
```

Comment:
You need to convert the customer string back to an object.
```suggestion
            try {
              currentCustomer = JSON.parse(customer);
              deferred.resolve(currentCustomer);
            } catch (error) {
              deferred.reject(error);
            }
```

Reply:
If the customer is null, the JSON.parse is working. So I prefer testing `customer != null` instead of adding a try catch.

Reply:
Finally, not test is needed due to the null object pattern. JSON parse will work anyway.